servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherProcessorBuffer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import io.servicetalk.concurrent.PublisherSource.Processor;
+
+/**
+ * A buffer to store items for a {@link Processor}.
+ *
+ * @param <T>  Type of items stored in this buffer.
+ */
+public interface PublisherProcessorBuffer<T> extends ProcessorBuffer<T> {
+
+    /**
+     * Try to consume the next item stored in this buffer. If there are no items stored in the buffer and the buffer has
+     * terminated {@link #terminate() successfully} or with an {@link #terminate(Throwable) error} then consume that
+     * {@link BufferConsumer#consumeTerminal() successful} or
+     * {@link BufferConsumer#consumeTerminal(Throwable) failed} termination.
```

Comment:
`BufferConsumer` implementations that exist today do not support concurrent access, and it would be surprising to have concurrency consumption, is this something we expect when the consumer is typically a `Subscriber`? The connection to the `Processor` comments is nuanced and I think it is worth clarifying here to avoid confusion (see https://github.com/apple/servicetalk/pull/1039#discussion_r421359876, https://github.com/apple/servicetalk/pull/1039#discussion_r421365637).